govcd/catalog_test.go

```diff
@@ -84,3 +89,181 @@ func (vcd *TestVCD) Test_DeleteCatalog(check *C) {
+
+	//execute
+	uploadTask, err := catalog.UploadOvf(vcd.config.OVA.OVAPath, itemName, "upload from test", 1024)
+	check.Assert(err, IsNil)
+
+	//take control of stdout
+	oldStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	uploadTask.ShowUploadProgress()
+
+	w.Close()
+	//read stdin
+	result, _ := ioutil.ReadAll(r)
+	os.Stdout = oldStdout
+
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	check.Assert(string(result), Matches, ".*Upload progress 100.00%")
+
+	catalog, err = org.FindCatalog(vcd.config.VCD.Catalog.Name)
+	verifyCatalogItemUploaded(check, catalog, itemName)
+}
+
+// Tests System function UploadOvf by creating catalog, creating catalog item
+// and expexting specific error then trying to create same catalog item. As vCD returns cryptic error for such case.
+func (vcd *TestVCD) Test_UploadOvf_error_withSameItem(check *C) {
+	skipThenOvaPathMissing(vcd, check)
+
+	itemName := TestUploadOvf + "5"
+
+	catalog, _ := findCatalog(vcd, check, vcd.config.VCD.Catalog.Name)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_UploadOvf")
+
+	//add item
+	uploadTask, err2 := catalog.UploadOvf(vcd.config.OVA.OVAPath, itemName, "upload from test", 1024)
+	check.Assert(err2, IsNil)
+	err2 = uploadTask.WaitTaskCompletion()
+	check.Assert(err2, IsNil)
+
+	catalog, _ = findCatalog(vcd, check, vcd.config.VCD.Catalog.Name)
+	_, err3 := catalog.UploadOvf(vcd.config.OVA.OVAPath, itemName, "upload from test", 1024)
+	check.Assert(err3.Error(), Matches, ".*already exists. Upload with different name.*")
+}
+
+// Tests System function UploadOvf by creating catalog, uploading file and verifying
+// that extracted files were deleted.s
+func (vcd *TestVCD) Test_UploadOvf_cleaned_extracted_files(check *C) {
+	skipThenOvaPathMissing(vcd, check)
+
+	itemName := TestUploadOvf + "6"
+
+	//check existing count of folders
+	oldFolderCount := countFolders()
+
+	catalog, _ := findCatalog(vcd, check, vcd.config.VCD.Catalog.Name)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_UploadOvf")
+
+	uploadTask, err := catalog.UploadOvf(vcd.config.OVA.OVAPath, itemName, "upload from test", 1024)
+	check.Assert(err, IsNil)
+	err = uploadTask.WaitTaskCompletion()
+	check.Assert(err, IsNil)
+
+	check.Assert(oldFolderCount, Equals, countFolders())
+
+}
+
+func countFolders() int {
+	files, err := ioutil.ReadDir(os.TempDir())
+	if err != nil {
+		log.Fatal(err)
+	}
+	count := 0
+	for _, f := range files {
+		if strings.Contains(f.Name(), util.TmpDirPrefix+".*") {
+			count++
+		}
+	}
+	return count
+}
+
+func checkUploadOvf(vcd *TestVCD, check *C, ovaFileName, catalogName, itemName string) {
+	//setupCatalog(vcd, check, testCreateCatalog)
+
+	catalog, org := findCatalog(vcd, check, vcd.config.VCD.Catalog.Name)
+	AddToCleanupList(itemName, "catalogItem", vcd.org.Org.Name+"|"+vcd.config.VCD.Catalog.Name, "Test_UploadOvf")
```

Comment:
The call to `AddToCleanupList` should only happen *after* the successful creation of an item.